components/tidb_query_normal_expr/src/builtin_time.rs

```diff
@@ -1824,4 +1850,28 @@ mod tests {
         // test NULL case
         test_err_case_one_arg(&mut ctx, ScalarFuncSig::Month, Datum::Null);
     }
+
+    #[test]
+    fn test_make_date() {
+        let cases = vec![
+            (2014, 224234, "2627-12-07"),
+            (2014, 1, "2014-01-01"),
+            (7900, 705000, "9830-03-23"),
+            (7901, 705000, "9831-03-23"),
+            (7904, 705000, "9834-03-22"),
+            (8000, 705000, "9930-03-23"),
+            (8001, 705000, "9931-03-24"),
+        ];
+        let mut ctx = EvalContext::default();
+        for (arg1, arg2, exp) in cases {
+            let datetime = Time::parse_date(&mut ctx, exp).unwrap();
+            test_ok_case_two_arg(
+                &mut ctx,
+                ScalarFuncSig::MakeDate,
+                Datum::I64(arg1),
+                Datum::I64(arg2),
+                Datum::Time(datetime),
+            );
+        }
+    }
```

Comment:
Could you also add some test cases that outputs null? I provide some test cases, some of them fail with your implementation:
```rust
let null_cases = vec![
    (Datum::Null, Datum::Null),
    (Datum::I64(2014), Datum::I64(0)),
    (Datum::I64(10000), Datum::I64(1)),
    (Datum::I64(9999), Datum::I64(366)),
    (Datum::I64(-1), Datum::I64(1)),
    (Datum::I64(-4294965282), Datum::I64(1)),
];
for (arg1, arg2) in null_cases {
    test_err_case_two_arg(&mut ctx, ScalarFuncSig::MakeDate, arg1, arg2);
}
```

Reply:
@trabbart I suggest that we stick on the implementation from `TiDB`: https://github.com/pingcap/tidb/blob/9d391949cdc4b513c682063113a79dfdd471497f/expression/builtin_time.go#L5724

Reply:
I added test cases that outputs null. Now code is passing new test casses. 

I've changed code to look more similar to TiDB implementation.